python/taichi/lang/ast/ast_transformer.py

```diff
@@ -515,6 +515,10 @@ def build_Call(ctx, node):
         if ASTTransformer.build_call_if_is_type(ctx, node, args, keywords):
             return node.ptr
 
+        if getattr(node.func, 'call_tensor_op', False):
```

Comment:
why dont we directly check if `caller` method exists?

Reply:
True!